exonum-java-binding/core/rust/src/proxy/node.rs

```diff
@@ -130,7 +130,12 @@ pub extern "system" fn Java_com_exonum_binding_service_NodeProxy_nativeSubmit(
                 match node.submit(raw_transaction) {
                     Ok(tx_hash) => convert_hash(&env, &tx_hash),
                     Err(err) => {
-                        let error_class = INTERNAL_SERVER_ERROR;
+                        // node#submit can fail for two reasons: unknown transaction id and
+                        // an error in ApiSender#send. The former is the service implementation
+                        // error and is appropriate to communicate with the exception below;
+                        // the second is an internal, unrecoverable error that has nothing
+                        // to do with the service: ECR-3190
```

Comment:
What is expected from ECR-3190 after the completion of this PR? From its current description one could conclude that this PR also solves ECR-3190

Reply:
yes, updated.